src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/NuGetProjectTelemetryService.cs

```diff
@@ -64,21 +64,21 @@ public void EmitNuGetProject(NuGetProject nuGetProject)
 
                 // Get project type.
                 var projectType = NuGetProjectType.Unknown;
-                if (nuGetProject is MSBuildNuGetProject)
+                if (nuGetProject.ProjectStyle == ProjectModel.ProjectStyle.PackagesConfig)
                 {
```

Comment:
This PR doesn't claim to make everything consistent. These changes are made to avoid loading deferred projects where it's possible.